homeassistant/components/pushover/config_flow.py

```diff
@@ -87,9 +87,13 @@ async def async_step_user(
         errors = {}
 
         if user_input is not None:
-            await self.async_set_unique_id(user_input[CONF_USER_KEY])
-            self._abort_if_unique_id_configured()
```

Comment:
So the combination of user key and API key is unique right?

Shouldn't that be used as the unique identifier instead?

Reply:
I first thought of that. But if the api key is removed the user can reauthenticate with a new api key. So that makes the api key not unique.
Can we update the unique_id when reauthenticating?

Reply:
As I read the documentation of the Pushover API the API token is unique. Wouldn't that solve the problem? If a user registers this in HA it would be a new instance. Just make the API not changeable for an existing configuration but document the procedure to change the API key is create a new configuration?

Reply:
An API key can be regenerated which means for the same application an old api token can be revoked and new one used instead.
![image](https://user-images.githubusercontent.com/5123967/189301342-302c56eb-7761-4744-800d-a231e44bf276.png)
This means we should still provide a re-authentication flow to enter new api_key.

Reply:
@engrbm87 Can I help you in any way? If a re-authentication flow is problematic I think a valid scenario for now would be to consider the API Token as unique and document the need to remove and recreate the configuration through the interface.

Reply:
Hello @frenck, appreciate your thoughts on this. How do you recommend to proceed.

Reply:
I also found out you can reset the User Key in Pushover, which presents an issue with the current unique_id. Since there is no return value from the API like a name I think there is no way to create a unique_id based on the configuration data. The portal uses the App Key as a unique referrer so I think this would be best unique_id at the moment. As I think it is an edge case that a user switches an App Key (and even less a change of the User Key) I would suggest using the api_key instead of the user_key. The user key can be made a 'global' configuration item for the integration (which could be changed without impact to existing instances) and if the user needs to change the App Key this requires the deletion of the old instance and the creation of a new one, possibly with the same name. This _should_ have no impact on automations.

Reply:
Or just create a GUID and use that as the unique_id. It is not connected to any combination of User Key and App Key and therefore you can reconfigure both without adjusting the unique_id. I feel this is not the right choice because there would be nothing stopping you from creating multiple services with the same User/App key pair. On the other hand, this does not break anything, just the behavior on the phone-app might not be wat the user expects.

Reply:
> I feel this is not the right choice because there would be nothing stopping you from creating multiple services with the same User/App key pair. On the other hand, this does not break anything, just the behavior on the phone-app might not be wat the user expects.

notifying oneself with the same user/app key pair is something you can do with Pushover within other services already, so i don't really see the problem with allowing it here. that said, if a way to keep bad/duplicated data down to a minimum is available, i feel like that option is superior.

Reply:
> Or just create a GUID and use that as the unique_id. It is not connected to any combination of User Key and App Key and therefore you can reconfigure both without adjusting the unique_id. I feel this is not the right choice because there would be nothing stopping you from creating multiple services with the same User/App key pair. On the other hand, this does not break anything, just the behavior on the phone-app might not be wat the user expects.

i know i'm a fringe user as this issue affects me to begin with, but i think i might be 'really' fringe user as i have(had until the HA update) two different Pushover User Keys defined. one for personal use and one for work (Group Key) that gets used by the team. locking the User Key to a single instance would still be a breaking change from the YAML config option that was originally available.

Reply:
I think it's fine to reject adding another config entry with the same user_key + api_key as implemented in the PR.
However, should that check also be done in the reauth flow?

Reply:
I added the check in the reauth. Please check and confirm.

Reply:
Many thanks for progress here!
Do the code changes handle only creating new entries by GUI or would it also allow importing multiple entries from existing `configuration.yaml`?